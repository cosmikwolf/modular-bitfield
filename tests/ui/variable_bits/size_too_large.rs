use modular_bitfield::prelude::*;

#[derive(Specifier)]
#[variable_bits(8, 256)]  // 256 > 128 bits max
enum TestData {
    #[discriminant = 0]
    Small(u8),
    #[discriminant = 1]
    TooLarge,
}

fn main() {}