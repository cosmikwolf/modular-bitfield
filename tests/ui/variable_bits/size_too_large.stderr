error: variable_bits sizes cannot exceed 128 bits
 --> tests/ui/variable_bits/size_too_large.rs:4:20
  |
4 | #[variable_bits(8, 256)]  // 256 > 128 bits max
  |                    ^^^
